packages/toolkit/src/amazonqGumby/activation.ts

```diff
@@ -13,14 +13,43 @@ import { transformByQState } from '../codewhisperer/models/model'
 import * as CodeWhispererConstants from '../codewhisperer/models/constants'
 import { ProposedTransformationExplorer } from '../codewhisperer/service/transformationResultsViewProvider'
 import { codeTransformTelemetryState } from './telemetry/codeTransformTelemetryState'
-import { telemetry } from '../shared/telemetry/telemetry'
-import { CancelActionPositions, logCodeTransformInitiatedMetric } from './telemetry/codeTransformTelemetry'
+import { CodeTransformJavaSourceVersionsAllowed, telemetry } from '../shared/telemetry/telemetry'
+import {
+    CancelActionPositions,
+    JDKToTelemetryValue,
+    logCodeTransformInitiatedMetric,
+} from './telemetry/codeTransformTelemetry'
 import { CodeTransformConstants } from './models/constants'
+import { getOpenProjects, validateOpenProjects } from '../codewhisperer/service/transformByQHandler'
+import { MetadataResult } from '../shared/telemetry/telemetryClient'
 
 export async function activate(context: ExtContext) {
     const transformationHubViewProvider = new TransformationHubViewProvider()
     new ProposedTransformationExplorer(context.extensionContext)
 
+    // Register an activation event listener to determine when the IDE opens, closes or users
+    // select to open a new workspace
+    const workspaceChangeEvent = vscode.workspace.onDidChangeWorkspaceFolders(event => {
+        // A loophole to register the IDE closed. This is when no folders were added nor
+        // removed, but the event still fired. This assumes the user closed the workspace
+        if (event.added.length === 0 && event.removed.length === 0) {
+            // Only fire closed during running/active job status
+            if (transformByQState.isRunning()) {
+                telemetry.codeTransform_jobIsClosedDuringIdeRun.emit({
+                    codeTransformJobId: transformByQState.getJobId(),
+                    codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+                    codeTransformStatus: transformByQState.getStatus(),
+                })
+            }
+        } else {
+            telemetry.codeTransform_jobIsResumedAfterIdeClose.emit({
```

Comment:
gumby's activate() is always called, which means this will always emit even for users that have never used gumby. does that make sense?

Reply:
Maybe we can reduce the logic here to fire less. To just codewhisperer users only or some form of eligible users.

Reply:
Won't this fire when a user opens a new folder or removes one?

Reply:
Basically I don't know that we even need this `else` block

Reply:
We will already have the info we need if the user actually clicks on "Transform", so what we are more so looking for is info on the users who do not click Transform (specifically, do they have a project valid for transformation open in their workspace?). One thing we could do to somewhat narrow things down is at least add this condition (https://github.com/aws/aws-toolkit-vscode/blob/5faec4ac40181605e6e8611e2a381310265873f7/packages/toolkit/src/amazonq/explorer/amazonQNode.ts#L87) before emitting the metric since that tells us the "Transform" button is visible to the user.

Reply:
If we really wanted to restrict metric emission to just users who have clicked "Transform", we can simply emit a metric here if `validProjects.length > 0`:
https://github.com/aws/aws-toolkit-vscode/blob/5faec4ac40181605e6e8611e2a381310265873f7/packages/toolkit/src/codewhisperer/commands/startTransformByQ.ts#L409